### ISSUE TITLE
Refactor JDKPacakgerPlugin dependencies

### ISSUE THREAD

**Issue by muuki88** [sbt/sbt-native-packager#753]
All packaging plugins (Debian, Rpm, Docker, etc.) require only other packaging plugins (e.g. Rpm requires Linux). The `JDKPackager` plugin however requires two archetype plugins. In favor of consistency and flexibility we should remove these requirements and replace them with `UniversalPlugin`, which is the true dependency.

**Comment by metasim** [comment]
@muuki88 Is this what you're talking about?:

https://github.com/sbt/sbt-native-packager/blob/master/src/main/scala/com/typesafe/sbt/packager/jdkpackager/JDKPackagerPlugin.scala#L31

If so, what's the problem with having multiple up-stream plugins?

**Comment by muuki88** [comment]
Yes, that's the line. The thing is, that jdkpackager, for me, is a packaging format. And formats shouldn't make any assumptions about the app they package. However if we argue that jdkpackager is an archetype, this dependency would be OK.

I'm not sure on this. I discovered this inconsistency during the docs refactoring, where it's listed as a packaging format.

**Comment by metasim** [comment]
In some ways it's a hybrid, perhaps closer to an archetype. Because it will generate so many platform specific executable forms, including Linux/Windows/MacOS it's kinda an archetype. But then from that (if you want) it will generate multiple packages: .msi, .dmg, .deb, .rpm, etc., making it more akin to a format.

So it's kinda an oddball. 

It being in the packaging format section of the docs hides it a bit, so selfishly I'd love to see it moved to the  archetype side of the house. Thoughts?

**Comment by muuki88** [comment]
Hm, generating packages is a sign for a packaging plugin ;) No matter how many packages it will generate. For me, it has too many package specific properties

- defines it's own scope `jdkpackager`
- implements the `packageBin` task
- creates packages
- introduces package-tool specific tasks
- doesn't change existing settings

The change I would like to do is

- `JavaAppPackaging` enables the `JDKPackagerPlugin`
- `JDKPackagerPlugin` requires only the `UniversalPlugin`

This will allow

- Use jdkpackaging with other archetypes
- Choose `LauncharJar` or `ClasspathJar` or none of them
- consistency with the other packaging plugins

I understand JDKPackager is not the "classic" packaging plugin, which provides _one_ format. However making it an alternative, doesn't mean it's an archetype. `JDebPackaging` is still a packaging plugin, even it provides a different debian-packaging implementation. 

The usage of the `JDKPackagerPlugin` would also be streamlined with the other formats.

```scala
enablePlugins(JavaAppPackaging, LauncherJarPlugin)
// package specific settings
```

done. I really would like to make that shift.

**Comment by metasim** [comment]
@muuki88 I'm sorry, but I'm not really following what needs to be done; maybe I just don't have enough context. Do you have an example of a similar restructuring?
